app/controllers/applications_controller.rb

```diff
@@ -150,6 +150,20 @@ def application_params
   end
 
   def dashboard_url(host_name, application_name)
+    grafana_base_url = Plek.find('grafana')
+    uri = URI("#{grafana_base_url}/api/dashboards/file/deployment_#{application_name}.json")
+
+    begin
+      grafana_api_response = Net::HTTP.get_response(uri)
```

Comment:
I've moved the check so it only happens once.

The Net:HTTP default timeout is 60 seconds. I've reduced it to 2 seconds. It's generally much faster than that because the endpoint just returns the dashboard JSON - it doesn't have to load any application data.

What do you think about the behaviour in development. At the moment we always rescue an exception in dev because Grafana is not running there, which means the link never appears. Would it be better to always show it in dev?